e-library/src/Components/forms/createbook.jsx

```diff
@@ -1,29 +1,69 @@
-import React from "react";
+import React, { useState } from "react";
+import { useNavigate } from 'react-router-dom';
+import axios from "../api/axios";
+import formData from 'form-data'
+import { token } from './input';
 
-const CreateBook = () => {
-    return(
-        <form >
-        <h2>Create your book</h2>
-        <div className="input-box">
-            <input type="text" placeholder="Title"/>
-        </div>
-        <div className="input-box">
-           <textarea placeholder="Description" rows="4" columns="50"></textarea>
-        </div>
-        <div className="input-box">
-            <input type="text" placeholder="Author"/>
-        </div>
-        <div className="input-box">
-            <input type="text" placeholder="Price"/>
-        </div>
-        <div className="input-box">
-            <input type="file"/>
-        </div>
-        <div className="input-box">
-            <input type="submit" value="Create your book"/>
-        </div>
-    </form>
-    )
+const CreateBook = () => {;
+    const [title, setTitle] = useState('');
+    const [price, setPrice] = useState('')
+    const [author, setAuthor] = useState('');
+    const [fileUrl, setFileUrl] = useState(null)
+    const [description, setDescription] = useState('')
+
+    const redirect = useNavigate()
+
+    const submit = async (e) => {
+        e.preventDefault()
+        const form = new formData()
+        form.append('title', title)
+        form.append('price', price)
+        form.append('author', author)
+        form.append('description', description)
+        form.append('bookFile', fileUrl)
+        try {
+            const response = await axios.post('/api/books', form, {
+                headers: {
+                    'Content-Type': 'multipart/form-data',
+                    'authorization': token.id
+                }
+            });
+            setTitle('');
+            setPrice('');
+            setFileUrl('');
+            setAuthor('');
+            setDescription('');
+            window.alert('Book has been created successfully')
+            redirect('/homepage')
+            console.log(response.data)
+        } catch (err){
+              if (err) {
+                  window.alert('Failed to create book. Please try again')
+                  redirect('/homepage')
+            }
+        }
+    }
```

Comment:
refactor based on comments above. you're using state indiscriminately therefore why you have to setState multiple times.